tools/notification-configuration/github-codeowner-subscriber/CodeOwnersParser.cs

```diff
@@ -34,17 +34,25 @@ public CodeOwnersParser(string fileContents)
         /// </summary>
         /// <param name="path">Path to search in CODEOWNERS</param>
         /// <returns>A list of contacts at the associated path or an empty list if no contacts are found</returns>
-        public List<string> GetContactsForPath(string path)
+        public List<string> GetContactsForPath(string filePath)
         {
-            var result = expressionContacts
-                .Where(expr => path.StartsWith(expr.GlobExpression))
+            var pathResults = expressionContacts
+                .Where(expr => filePath.Equals(expr.GlobExpression, StringComparison.OrdinalIgnoreCase));
+
+            // If we find an exact file path match we should not find any other fallback matches based on path
+            if (!pathResults.Any())
+            {
+                var matchPath = PathWithoutFilename(filePath);
+                var searchPath = $"/{matchPath}/";
+
+                pathResults = expressionContacts
+                    .Where(expr => searchPath.StartsWith(expr.GlobExpression));
```

Comment:
Should probably be ignoring case here, too

```suggestion
                    .Where(expr => searchPath.StartsWith(expr.GlobExpression, StringComparison.OrdinalIgnoreCase));
```

Reply:
Yes I will.